lib/viewmodels/user_comments_view_model.dart

```diff
@@ -0,0 +1,71 @@
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/comment/comment_id_firestore.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/database/user_comment/user_comment_id_firestore.dart";
+import "package:proxima/models/ui/user_comment_details.dart";
+import "package:proxima/services/database/comment_repository_service.dart";
+import "package:proxima/services/database/user_comment_repository_service.dart";
+import "package:proxima/viewmodels/login_view_model.dart";
+import "package:proxima/viewmodels/posts_feed_view_model.dart";
+
+typedef UserCommentsState = List<UserCommentDetails>;
+
+class UserCommentViewModel extends AutoDisposeAsyncNotifier<UserCommentsState> {
+  UserCommentViewModel();
+
+  @override
+  Future<UserCommentsState> build() async {
+    final userCommentRepository =
+        ref.watch(userCommentRepositoryServiceProvider);
+    final user = ref.watch(validLoggedInUserIdProvider);
+
+    final commentsFirestore = await userCommentRepository.getUserComments(user);
+    final comments = commentsFirestore.map((comment) {
+      final userComment = UserCommentDetails(
+        userCommentId: comment.id,
+        commentId: comment.data.commentId,
+        parentPostId: comment.data.parentPostId,
+        description: comment.data.content,
+      );
+      return userComment;
+    }).toList();
+
+    return comments;
+  }
+
+  /// Delete the comment with the given [commentId] from the database
+  /// and refresh the state of this viewmodel (list of user comments).
+  Future<void> deleteComment(
+    UserCommentIdFirestore userComment,
```

Comment:
Renaming this to `userCommentId` would be more consistent I think.

Reply:
Fixed with [d93f38d](https://github.com/ProximaEPFL/proxima/pull/269/commits/d93f38d58c5b6c3fbe5639ab08be28f5d2fea604)